cleverhans/experimental/certification/dual_formulation.py

```diff
@@ -426,61 +423,43 @@ def get_lanczos_eig(self, compute_m=True, feed_dict=None):
 
-  def compute_certificate(self, current_step):
+  def compute_certificate(self, current_step, feed_dict):
     """ Function to compute the certificate based either current value
     or dual variables loaded from dual folder """
-    lambda_neg_val = self.sess.run(self.lambda_neg)
-    lambda_lu_val = self.sess.run(self.lambda_lu)
-
-    # Use 1000 iterations for Lanczos in compute certificate
-    m_guess_vec = np.zeros((self.matrix_m_dimension, 1))
-    feed_dict = {
-        self.h_min_vec_ph: self.h_min_vec_estimate,
-        self.m_min_vec_ph: m_guess_vec
-    }
-    _, min_eig_val_h = self.get_lanczos_eig(compute_m=False, feed_dict=feed_dict)
+    nu = feed_dict[self.nu]
+    _, second_term = self.make_m_psd(nu, feed_dict)
+    tf.logging.info("Nu after modifying: " + str(second_term))
+    feed_dict.update({self.nu: second_term})
 
-    new_lambda_lu_val = [np.copy(x) for x in lambda_lu_val]
-    new_lambda_neg_val = [np.copy(x) for x in lambda_neg_val]
-
-    for i in range(self.nn_params.num_hidden_layers + 1):
-      # Making H PSD
-      new_lambda_lu_val[i] = lambda_lu_val[i] + 0.5*np.maximum(-min_eig_val_h, 0) + TOL
-      # Adjusting the value of \lambda_neg to make change in g small
-      new_lambda_neg_val[i] = lambda_neg_val[i] + np.multiply((self.lower[i] + self.upper[i]),
-                                                              (lambda_lu_val[i] -
-                                                               new_lambda_lu_val[i]))
-      new_lambda_neg_val[i] = (np.multiply(self.negative_indices[i],
-                                           new_lambda_neg_val[i]) +
-                               np.multiply(self.switch_indices[i],
-                                           np.maximum(new_lambda_neg_val[i], 0)))
-
-    # Assign new lambda
-    # TODO(shankarshreya): remove the assign ops and do this with new variables
-    self.sess.run([tf.assign(var, val) for var, val in zip(self.lambda_lu, new_lambda_lu_val)])
-    self.sess.run([tf.assign(var, val) for var, val in zip(self.lambda_neg, new_lambda_neg_val)])
-
-    # Make matrix M PSD
-    scalar_f = self.sess.run(self.scalar_f)
-    _, second_term = self.make_m_psd(feed_dict)
-
-    computed_certificate = scalar_f + 0.5*second_term
+    # Add 0.05 to final nu to account for numerical instability
+    computed_certificate = self.sess.run(self.unconstrained_objective, feed_dict=feed_dict)
 
     tf.logging.info('Inner step: %d, current value of certificate: %f',
                     current_step, computed_certificate)
 
     # Sometimes due to either overflow or instability in inverses,
     # the returned certificate is large and negative -- keeping a check
-    if LOWER_CERT_BOUND < computed_certificate < -1:
-      _, min_eig_val_m = self.get_lanczos_eig()
+    if LOWER_CERT_BOUND < computed_certificate < 0:
+      _, min_eig_val_m = self.get_lanczos_eig(feed_dict=feed_dict)
+      tf.logging.info("min eig val from lanczos: " + str(min_eig_val_m))
```

Comment:
please update quote style to use `'` to be consistent with the rest of the codebase.
Also consider removing this logging line if this logging is not needed.